File: Q38.py
#Q38.py
class Solution:
    def countAndSay(self, n: int) -> str:
        s = "1"
        for _ in range(n-1):
            t = []
            i = 0
            L = len(s)
            while i < L:
                j = i+1
                while j < L and s[j] == s[i]:
                    j += 1
                t.append(str(j-i))
                t.append(s[i])
                i = j
            s = "".join(t)
        return s

File: Q38.cpp
//Q38.cpp
// C++ — fastest / minimal allocations
class Solution {
public:
    string countAndSay(int n) {
        string s = "1";
        while(--n){
            string t;
            t.reserve(s.size()*2);
            for(int i=0;i<s.size();){
                int j = i;
                while(j<s.size() && s[j]==s[i]) j++;
                t += char('0' + (j-i));
                t += s[i];
                i = j;
            }
            s.swap(t);
        }
        return s;
    }
};
